vshard/replicaset.lua

```diff
@@ -1198,8 +1201,13 @@ replicaset_mt.__index = index
 local replica_mt = {
     __index = {
         is_connected = function(replica)
-            return replica.conn and replica.conn:is_connected() and
-                conn_vconnect_check_or_close(replica.conn)
+            if not replica.conn then
+                return false
+            end
+            if not replica.conn:is_connected() then
+                return false
+            end
+            return conn_vconnect_check_or_close(replica.conn)
         end,
```

Comment:
I get your point about returning an error for reporting, but typically functions called with `is_...` prefix should only return true or false. Any other behaviour is highly unexpected. If you want an error to be returned, I would suggest  to introduce a new function `check_is_connected()` which would work the way you need. And `is_connected()` should be left with true/false result.

Reply:
Sure, added the `check_is_connected`, left `is_connected` intact.

I also return err even when there's no `conn` at all. I suppose, it's ok to overwrite prev error even if it was from `vconnect`. Both these situations just mean, that the connection is not done yet. However, returning just `last_err` is not a good approach, IMO, it should be a map of errors.

Reply:
> I also return err even when there's no conn at all

I specifically wanted to avoid it as it is not technically an error. Just nobody talked to this instance yet. But I am fine with the new behaviour. If somebody would complain about too many error logs, you can do another PR to drop this error.

> However, returning just last_err is not a good approach, IMO, it should be a map of errors.

Doesn't many too much sense, I think. The reason for one error was that people would fix those errors one by one if they have multiple. Many errors are likely to be independent anyway because would be about different instances. Didn't want to complicate the code. Again, if anybody asks, we can always change that.